.lighthouserc.js

```diff
@@ -0,0 +1,30 @@
+// https://github.com/GoogleChrome/lighthouse-ci/blob/main/docs/configuration.md
+module.exports = {
+  ci: {
+    collect: {
+      numberOfRuns: 1,
+      // Replace with correct URLs depending on context
+      url: [
+        'https://beta.uniprot.org',
```

Comment:
This is probably my lack of knowledge - are we testing the build or the actual live site?

Reply:
To run the full audit, we have to test against an actual deploy of the website, could be internal URL though (available through some Gitlab environment variable?).
This is because it's also testing server settings (like server compression, and stuff like that).
That's why I thought it should run only on the main branch, *after* it has been deployed.

We could have an other subset of tests (like for accessibility and some of the tests that are not depending on the server) to run on a localhost server just running for the job. That could allow us to test at least a subset of audits and make the build fail if this subset fails.

So yeah, basically I think we'd need 2 kinds of lighthouse audit, full, or subset.

Reply:
It would be good to test against the internal URL as this would mean we wouldn't test the EBI routing infrastructure as part of the tests.
Good idea about the subset, although we'd have to carefully chose what we want to include in there.

Reply:
To be fair, I think the point of some of the audits here is to test as close as possible to what a user would see, so the "full lighthouse" I believe should be run on the public-facing URL. Because if we get a "First Contentful Paint" of "1.5s" internally we might think it's fine, but if for a real user it's "20s" we have a problem to tackle, either with the EBI routing, or optimising even more on our side, or both.
But I'd say we'd want to know for sure!

Reply:
While it's true that we need to test the "whole experience" maybe it is not something we want to check with every push to master, while something we did in the code which would affect the performance is.